src/CommunityToolkit.Maui/Behaviors/PlatformBehaviors/Touch/GestureManager.cs

```diff
@@ -0,0 +1,789 @@
+			return;
+		}
+		do
+		{
+			var rippleState = isToggled.HasValue && isToggled.Value
+				? TouchState.Normal
+				: TouchState.Pressed;
+
+			await RunAnimationTask(sender, rippleState, hoverState, animationTokenSource.Token);
+			if (token.IsCancellationRequested)
+			{
+				return;
+			}
+
+			rippleState = isToggled.HasValue && isToggled.Value
+				? TouchState.Pressed
+				: TouchState.Normal;
+
+			await RunAnimationTask(sender, rippleState, hoverState, animationTokenSource.Token);
+			if (token.IsCancellationRequested)
+			{
+				return;
+			}
+		}
+		while (--pulseCount != 0);
+	}
+
+	internal void HandleLongPress(TouchBehavior sender)
+	{
+		if (sender.State == TouchState.Normal)
+		{
+			longPressTokenSource?.Cancel();
+			longPressTokenSource?.Dispose();
+			longPressTokenSource = null;
+			return;
+		}
+
+		if (sender.LongPressCommand is null || sender.InteractionStatus == TouchInteractionStatus.Completed)
+		{
+			return;
+		}
+
+		longPressTokenSource = new CancellationTokenSource();
+		Task.Delay(sender.LongPressDuration, longPressTokenSource.Token).ContinueWith(t =>
+		{
+			if (t.IsFaulted && t.Exception != null)
+			{
+				throw t.Exception;
+			}
+
+			if (t.IsCanceled)
+			{
+				return;
+			}
+
+			var longPressAction = new Action(() =>
+			{
+				sender.HandleUserInteraction(TouchInteractionStatus.Completed);
+				sender.RaiseLongPressCompleted();
+			});
+			
+			if (sender.Dispatcher.IsDispatchRequired)
+			{
+				sender.Dispatcher.Dispatch(longPressAction);
+			}
+			else
+			{
+				longPressAction.Invoke();
+			}
+		});
+	}
+
+	internal void SetCustomAnimationTask(Func<TouchBehavior, TouchState, HoverState, int, Easing?, CancellationToken, Task>? animationTaskFactory)
+		=> this.animationTaskFactory = animationTaskFactory;
+
+	internal void Reset()
+	{
+		SetCustomAnimationTask(null);
+		defaultBackgroundColor = default;
+	}
+
+	internal void OnTapped(TouchBehavior sender)
+	{
+		if (!sender.CanExecute || (sender.LongPressCommand != null && sender.InteractionStatus == TouchInteractionStatus.Completed))
+		{
+			return;
+		}
+
+		if (DeviceInfo.Platform == DevicePlatform.Android)
+		{
+			HandleCollectionViewSelection(sender);
+		}
+
+		if (sender.Element is IButtonController button)
+		{
+			button.SendClicked();
+		}
+
+		sender.RaiseCompleted();
+	}
+
+	void HandleCollectionViewSelection(TouchBehavior sender)
+	{
+		CollectionView? parent = null;
+		VisualElement? result = null;
+		if (!sender.Element?.TryFindParentElementWithParentOfType(out result, out parent) ?? true)
+		{
+			return;
+		}
+
+		var collectionView = parent ?? throw new NullReferenceException();
+
+		var item = result?.BindingContext ?? result ?? throw new NullReferenceException();
+
+		switch (collectionView.SelectionMode)
+		{
+			case SelectionMode.Single:
+				collectionView.SelectedItem = item;
+				break;
+			case SelectionMode.Multiple:
+				var selectedItems = collectionView.SelectedItems?.ToList() ?? new List<object>();
```

Comment:
`SelectedItems` is already a `IList` so no need to call the `.ToList()` method.
```suggestion
				var selectedItems = collectionView.SelectedItems ?? [];
```

Reply:
Updated